ui/de-lib/src/main/java/org/iplantc/de/analysis/client/views/dialogs/AnalysisSharingDialog.java

```diff
@@ -0,0 +1,106 @@
+package org.iplantc.de.analysis.client.views.dialogs;
+
+import org.iplantc.de.analysis.client.presenter.sharing.AnalysisSharingPresenter;
+import org.iplantc.de.analysis.client.views.sharing.AnalysisSharingView;
+import org.iplantc.de.analysis.client.views.sharing.AnalysisSharingViewImpl;
+import org.iplantc.de.client.models.analysis.Analysis;
+import org.iplantc.de.client.services.AnalysisServiceFacade;
+import org.iplantc.de.client.sharing.SharingPresenter;
+import org.iplantc.de.client.util.JsonUtil;
+import org.iplantc.de.collaborators.client.util.CollaboratorsUtil;
+import org.iplantc.de.commons.client.views.dialogs.IPlantDialog;
+
+import com.google.common.base.Preconditions;
+import com.google.inject.Inject;
+
+import com.sencha.gxt.core.client.ValueProvider;
+import com.sencha.gxt.data.shared.ListStore;
+import com.sencha.gxt.data.shared.ModelKeyProvider;
+import com.sencha.gxt.widget.core.client.event.SelectEvent;
+import com.sencha.gxt.widget.core.client.event.SelectEvent.SelectHandler;
+import com.sencha.gxt.widget.core.client.grid.ColumnConfig;
+import com.sencha.gxt.widget.core.client.grid.ColumnModel;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AnalysisSharingDialog extends IPlantDialog implements SelectHandler {
+
+    private final AnalysisServiceFacade analysisService;
+
+    private SharingPresenter sharingPresenter;
+
+    @Inject
+    CollaboratorsUtil collaboratorsUtil;
+    @Inject
+    JsonUtil jsonUtil;
+
+    @Inject
+    AnalysisSharingDialog(final AnalysisServiceFacade analysisService) {
```

Comment:
Here, you could inject `AnalysisSharingPresenter`, and assign it to your `SharingPresenter` variable.
Same holds for the `DataSharingDialog` and `DataSharingPresenter`.